posthog/models/instance_setting.py

```diff
@@ -18,12 +23,30 @@ class Meta:
     def value(self):
         return json.loads(self.raw_value)
 
+    def delete(self, *args, **kwargs):
+        cache.delete(self.key.replace(CONSTANCE_DATABASE_PREFIX, ""))
+        super().delete(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        cache.set(self.key.replace(CONSTANCE_DATABASE_PREFIX, ""), self.raw_value)
```

Comment:
Is there a possibility that we set the cache but then the update in PG doesn't work? I know for instance there is an [issue updating the `RECORDINGS_TTL_WEEKS` ](https://posthogusers.slack.com/archives/C01GLBKHKQT/p1667813517164979).

I think there is a bigger refactor to be done here, but perhaps for this immediate case we could move the cache setting to after an explicit commit of the instance setting?

Happy to take the inconsistency here if it's too complicated/not clear what's going on.

Reply:
Makes sense, we can move this after the call to `super().save()`, as that will trigger the PG update (and if it fails then were not doing any error handling here so cache won't be set).

Reply:
Note that `save` will not actually COMMIT however, I think that is handled at the request lifecycle level but I could be wrong.

Reply:
AFAIK, by default Django should be running in autocommit mode, which would mean every single SQL query is committed immediately. But I haven't looked over our settings, or dug too much into this...

Reply:
Yeah `ATOMIC_REQUESTS` is `False` by default and we don't set it afaict.

So the issue that I linked to I believe is due to this Django rest framework [instance setting update endpoint](https://github.com/PostHog/posthog/blob/master/posthog/api/instance_settings.py#L65) being hit. Not sure if drf has some other atomic settings. I might have debugged this incorrectly initially also, so take with a pinch of salt.